src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/model/BuildMemory.java

```diff
@@ -406,6 +408,24 @@ public void setEntryFailureMessage(GerritTriggeredEvent event, Run r, String fai
     }
 
     /**
+     * Creates a snapshot clone of the current coordination memory status.
+     *
+     * @return the report
+     */
+    @Nonnull
+    public synchronized BuildMemoryReport report() {
+        BuildMemoryReport report = new BuildMemoryReport();
+        for (Map.Entry<GerritTriggeredEvent, MemoryImprint> entry : memory.entrySet()) {
+            List<Entry> triggered = new LinkedList<Entry>();
```

Comment:
Because an ArrayList preallocates memory that might not be needed, and when that memory is full it needs to allocate even more new memory and move that over.
According to what basic algorithm theory taught me in school; an ArrayList is only effective to use when you need to access specific indices in the list. Everywhere else it is ineffective.
Since this list will only be used to iterate over from start to end once or twice, a LinkedList is more effective to use.